pkg/objstore/objstore.go

```diff
@@ -246,6 +251,9 @@ func (b *metricBucket) Upload(ctx context.Context, name string, r io.Reader) err
 	err := b.bkt.Upload(ctx, name, r)
 	if err != nil {
 		b.opsFailures.WithLabelValues(op).Inc()
+	} else {
+		//TODO: Use SetToCurrentTime() once we update the Prometheus client_golang
+		b.lastSuccessfullUploadTime.Set(float64(time.Now().UnixNano()) / 1e9)
```

Comment:
Why not using same as `lastHeartbeat.Set(float64(time.Now().Unix()))` (sidecar.go:153) ?

Reply:
Because this is [how its done in the prometheus client_golang](https://github.com/prometheus/client_golang/blob/master/prometheus/gauge.go#L99)

Reply:
Cool, I agree that having `float64(time.Now().UnixNano()) / 1e9` is  more accurate and should be used. But are you sure that having `.Set(float64(time.Now().Unix()))` for some timestamp and having `Set(float64(time.Now().UnixNano()) / 1e9)` for other in the same project is ok? 

Can we in this PR either:
- move all to the same things
- update client_golang vendor?

Reply:
I've updated the code to consistenly use `float64(time.Now().UnixNano()) / 1e9` because I couldn't quickly find out how to update just the `client_golang` package after removing its contraint in the `Gopkg.lock`